.github/workflows/ci.yaml

```diff
@@ -147,6 +147,9 @@ jobs:
           name: codecov-kas-fleet-manager # optional
           fail_ci_if_error: true # optional (default = false)
           verbose: true # optional (default = false)
+      - name: Validate that kas-fleet-manager images build successfully.
+        if: github.event_name == 'pull_request'
+        run: make image/build
```

Comment:
The set of steps is a little bit strange here. This step builds the image through the Makefile
but then in main we also build the image later.

Reply:
Yes. The later stage build and push the image when a change happens on the `main` branch. 
Here we only want to check that the image can be built.

Reply:
We should revisit the organization of the CI workflows and jobs as the current organization does not seem ideal.
For example building the container image on PRs could be done in parallel without waiting to any task.

Reply:
+1 on that. I think most of the steps here can be separate jobs